x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -100,6 +102,14 @@ private void handleException(String actionType, RestRequest request, RestChannel
                 @Override
                 protected boolean skipStackTrace() { return restStatus == RestStatus.UNAUTHORIZED; }
 
+                @Override
+                public Map<String, List<String>> filterHeaders(Map<String, List<String>> headers) {
+                    if (headers.containsKey("Warning")) {
+                        return Maps.copyMapWithRemovedEntry(headers, "Warning");
+                    }
+                    return headers;
+                }
+
```

Comment:
Do we have a test for this anywhere?

Reply:
Added a test to verify Warnings are removed if authN fails.